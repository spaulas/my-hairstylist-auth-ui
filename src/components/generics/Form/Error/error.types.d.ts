export type ErrorProps = {
  showError?: boolean;
  message?: string;
};
