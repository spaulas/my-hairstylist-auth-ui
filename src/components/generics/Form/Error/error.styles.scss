@import "@styles/imports";

.error {
  @include form-error;
  color: $primary--red__medium;
  position: absolute;
  opacity: 0;
  top: 48px;
  left: 24px;
  margin: 4px 0 0;
  transition: all 0.2s ease;

  &__visible {
    top: 56px;
    opacity: 100%;
  }
}

@media only screen and (max-width: $tablet-landscape__max) {
  .error {
    left: 20px;

    &__visible {
      top: 50px;
    }
  }
}

@media only screen and (max-width: $medium-desktop__max) {
  .error {
    left: 16px;

    &__visible {
      top: 44px;
    }
  }
}
