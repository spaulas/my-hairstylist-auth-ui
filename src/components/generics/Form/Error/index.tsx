import { ErrorProps } from "./error.types";
import "./error.styles.scss";

const FormError = ({ showError, message }: ErrorProps): React.ReactElement => {
  return (
    <p className={`error ${showError ? "error__visible" : ""}`}>{message}</p>
  );
};

export default FormError;
