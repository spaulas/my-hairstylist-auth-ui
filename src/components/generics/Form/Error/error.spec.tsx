import "@testing-library/jest-dom/extend-expect";
import Error from ".";
import { render, screen } from "@testing-library/react";

describe("Error", () => {
  const props = {
    showError: false,
    message: "Message",
  };

  it("renders correctly", () => {
    const { container } = render(<Error {...props} />);
    expect(container).toMatchSnapshot();
  });

  describe("className", () => {
    it("should not have error__visible when !showError", () => {
      render(<Error {...props} />);
      const error = screen.getByText(props.message);

      expect(error).toHaveClass("error");
      expect(error).not.toHaveClass("error__visible");
    });

    it("should have error__visible when showError", () => {
      const showErrorProps = { ...props, showError: true };
      render(<Error {...showErrorProps} />);
      const error = screen.getByText(props.message);

      expect(error).toHaveClass("error");
      expect(error).toHaveClass("error__visible");
    });
  });
});
