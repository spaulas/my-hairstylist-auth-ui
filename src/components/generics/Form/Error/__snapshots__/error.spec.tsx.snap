// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<div>
  <p
    class="error "
  >
    Message
  </p>
</div>
`;
